src/borg/remote.py

```diff
@@ -306,6 +316,26 @@ def ssh_cmd(self, location):
             args.append('%s' % location.host)
         return args
 
+    def write_bandwith_limited(self, to_send):
+        if self.ratelimit:
+            current = time.monotonic()
+            if self.ratelimit_last + RATELIMIT_PERIOD <= current:
+                self.ratelimit_quota += self.ratelimit
+                if self.ratelimit_quota > 2 * self.ratelimit:
+                    self.ratelimit_quota = 2 * self.ratelimit
```

Comment:
instead of adding up and then limiting, you could also conditionally add up.
```
if q < 2 * lim:
    q += lim
```
it goes up to almost 3 that way, but guess that doesn't matter, right?